tardis/visualization/plot_util.py

```diff
@@ -60,3 +73,255 @@ def get_mid_point_idx(arr):
+            (e.g. Si I - V), or any combination of these (e.g. species_list = [Si II, Fe I-V, Ca])
+
+        """
+
+        if species_list is not None:
+            # check if there are any digits in the species list. If there are, then exit.
+            # species_list should only contain species in the Roman numeral
+            # format, e.g. Si II, and each ion must contain a space
+            if any(char.isdigit() for char in " ".join(species_list)) == True:
+                raise ValueError(
+                    "All species must be in Roman numeral form, e.g. Si II"
+                )
+            else:
+                full_species_list = []
+                for species in species_list:
+                    # check if a hyphen is present. If it is, then it indicates a
+                    # range of ions. Add each ion in that range to the list as a new entry
+                    if "-" in species:
+                        # split the string on spaces. First thing in the list is then the element
+                        element = species.split(" ")[0]
+                        # Next thing is the ion range
+                        # convert the requested ions into numerals
+                        first_ion_numeral = roman_to_int(
+                            species.split(" ")[-1].split("-")[0]
+                        )
+                        second_ion_numeral = roman_to_int(
+                            species.split(" ")[-1].split("-")[-1]
+                        )
+                        # add each ion between the two requested into the species list
+                        for ion_number in np.arange(
+                            first_ion_numeral, second_ion_numeral + 1
+                        ):
+                            full_species_list.append(
+                                f"{element} {int_to_roman(ion_number)}"
+                            )
+                    else:
+                        # Otherwise it's either an element or ion so just add to the list
+                        full_species_list.append(species)
+
+                # full_species_list is now a list containing each individual species requested
+                # e.g. it parses species_list = [Si I - V] into species_list = [Si I, Si II, Si III, Si IV, Si V]
+                self._full_species_list = full_species_list
+                requested_species_ids = []
+                keep_colour = []
+
+                # go through each of the requested species. Check whether it is
+                # an element or ion (ions have spaces). If it is an element,
+                # add all possible ions to the ions list. Otherwise just add
+                # the requested ion
+                for species in full_species_list:
+                    if " " in species:
+                        requested_species_ids.append(
+                            [
+                                species_string_to_tuple(species)[0] * 100
```

Comment:
why `* 100`

Reply:
so that the IDs are all 3 or 4 digit numbers - e.g. Ca II = 2001

Reply:
FYI, these are all in the current implementation of the SDEC plotter - all this PR does is move these functions to a separate file to prevent duplication.